webapp/actions/global_actions.jsx

```diff
@@ -436,6 +436,13 @@ export function emitLocalUserTypingEvent(channelId, parentId) {
     const t = Date.now();
     const membersInChannel = ChannelStore.getStats(channelId).member_count;
 
+    if (global.window.mm_license.IsLicensed === 'true' && global.window.mm_config.ExperimentalTownSquareIsReadOnly === 'true') {
```

Comment:
global and window are the same thing here so these only need to be `global.mm_license` and `global.mm_config`

Reply:
@hmhealey `global.window` is being used in other places in this same file. Should I refactor those, too?

Reply:
If you want to. I've been changing those as I go along, but I haven't bothered with lines I'm not touching.